plugins/device/rest.go

```diff
@@ -121,6 +123,66 @@ func deviceListHandler(w http.ResponseWriter, r *http.Request) {
 
 }
 
+func deviceEditStateHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	peerID := mux.Vars(r)["peerid"]
+	if uuid.Parse(peerID) == nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Invalid peer-id passed in url")
+		return
+	}
+
+	req := new(deviceapi.EditDeviceStateReq)
+	if err := restutils.UnmarshalRequest(r, req); err != nil {
+		logger.WithError(err).Error("Failed to Unmarshal request")
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrJSONParsingFailed)
+		return
+	}
+
+	deviceState, exists := deviceapi.DeviceStates[strings.ToLower(req.State)]
+	if !exists {
+		logger.Error("State provided in request doesnot match any state supported by device")
+		errMsg := fmt.Sprintf("Invalid state, Supported states are (%v)", getKeys(deviceapi.DeviceStates))
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errMsg)
+		return
+	}
+
+	txn, err := transaction.NewTxnWithLocks(ctx, peerID)
+
+	if err != nil {
```

Comment:
a new code has been merged in #869, which returns a status code and error from an error message.
need to update the changes in this PR.
```
if err != nil {
		status, err := restutils.ErrToStatusCode(err)
		restutils.SendHTTPError(ctx, w, status, err)
		return
}
```

Reply:
@Madhu-1 Have you handled log messages in https://github.com/gluster/glusterd2/pull/869. What if I want to provide a different log message for a different type of error?